tpu-provisioner/internal/cloud/gke_test.go

```diff
@@ -128,3 +135,208 @@ func Test_tpuMachineType(t *testing.T) {
+						"google.com/tpu-provisioner-pod-namespace":    "default",
+					},
+					MachineType:            "ct5p-hightpu-4t",
+					ShieldedInstanceConfig: &container.ShieldedInstanceConfig{EnableIntegrityMonitoring: true},
+					SecondaryBootDisks: []containerv1beta1.SecondaryBootDisk{
+						{
+							DiskImage: "projects/my-gcp-project/global/images/my-disk-image",
+							Mode:      "CONTAINER_IMAGE_CACHE",
+						},
+					},
+				},
+				InitialNodeCount:  512,
+				Locations:         []string{""},
+				Management:        &container.NodeManagement{AutoRepair: true, AutoUpgrade: true},
+				MaxPodsConstraint: &container.MaxPodsConstraint{MaxPodsPerNode: 15},
+				Name:              "test-pool",
+				PlacementPolicy:   &container.PlacementPolicy{TpuTopology: "8x16x16", Type: "COMPACT"},
+				UpgradeSettings:   &container.UpgradeSettings{MaxSurge: 1},
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.desc, func(t *testing.T) {
+			gke := &GKE{
+				ClusterContext: tc.gkeContext,
+			}
+			pod := &v1.Pod{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "v1",
+					Kind:       "Pod",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"alpha.jobset.sigs.k8s.io/exclusive-topology": "cloud.google.com/gke-nodepool",
+						"batch.kubernetes.io/job-completion-index":    "0",
+						"jobset.sigs.k8s.io/job-index":                "0",
+						"jobset.sigs.k8s.io/job-key":                  "random-key",
+						"jobset.sigs.k8s.io/jobset-name":              "jobset-test",
+						"jobset.sigs.k8s.io/replicatedjob-name":       "job-1",
+						"jobset.sigs.k8s.io/replicatedjob-replicas":   "1",
+						"jobset.sigs.k8s.io/restart-attempt":          "0",
+					},
+					Finalizers: []string{"batch.kubernetes.io/job-tracking"},
+					Labels: map[string]string{
+						"batch.kubernetes.io/controller-uid":        "8484279a-de52-4ca1-b01e-130fbded30fb",
+						"batch.kubernetes.io/job-name":              "jobset-test-job-1-0",
+						"controller-uid":                            "8484279a-de52-4ca1-b01e-130fbded30fb",
+						"job-name":                                  "jobset-test-job-1-0",
+						"jobset.sigs.k8s.io/job-index":              "0",
+						"jobset.sigs.k8s.io/job-key":                "random-key",
+						"jobset.sigs.k8s.io/jobset-name":            "jobset-test",
+						"jobset.sigs.k8s.io/replicatedjob-name":     "job-1",
+						"jobset.sigs.k8s.io/replicatedjob-replicas": "1",
+						"jobset.sigs.k8s.io/restart-attempt":        "0",
+					},
+					Name:      "job-test-6gfwq",
+					Namespace: "default",
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion:         "batch/v1",
+							Kind:               "Job",
+							UID:                "8484279a-de52-4ca1-b01e-130fbded30fb",
+							Name:               "jobset-test-job-1-0",
+							Controller:         &trueVar,
+							BlockOwnerDeletion: &trueVar,
+						},
+					},
+					GenerateName:    "jobset-test-job-1-0-0-",
```

Comment:
Why are both Name and GenerateName set? Shouldn't it be one or the other?

Reply:
Test locally, both are set, the generated name should be set by job contoller.